### ISSUE TITLE
[RPi 5] Partition size issue on a 2TB NVMe

### ISSUE THREAD

**Issue by arusabu** [libretro/Lakka-LibreELEC#1963]
I'm using a Raspberry Pi 5, a PineberryPi HatDrive! Bottom and a 2TB NVMe, which is my main and only one drive (I want to have os and rom on the same disk).

On the first boot, Lakka doesn't resize correctly partitions :
![0ynmqlvwwjgc1](https://github.com/libretro/Lakka-LibreELEC/assets/167005981/7be4635c-b318-4216-91cd-3555477e9e3e)
(screen taken under an older version of Lakka but the exact same problem occurs with the current version of Lakka, "Lakka (official): 5.0 (RPi5.aarch64)")

The "LAKKA_DISK" partition can't be expanded to 1.82Tio with gparted by using resize2fs in Gparted ("New size results in too many block group descriptors.").

For this 2nd partition, do you use ex4 with 1KiB block size ?
If so, that's the/my issue.

As a workaround, once the NVMe is flashed and before using it, I use gparted to resize the 2nd partition to 2Gio and to create a 3rd one in ExFAT by using the remaining space.
![image](https://github.com/libretro/Lakka-LibreELEC/assets/167005981/1048608c-41c4-42b4-8789-312904809040)

**Comment by ToKe79** [comment]
the partition at creation time is 32 MiB, so `mke2fs` determines automatically small block size, most probably 1 KiB. i think we can go with 4 KiB blocks, so we can add `-b 4096` to `mke2fs`.

**Comment by arusabu** [comment]
Unless I'm doing something wrong, this issue is still presents on the devel-20240415170722-afaeeac (RPi5.aarch64)

After the 1st boot, I'm getting a new error :
![IMG_20240415_234230](https://github.com/libretro/Lakka-LibreELEC/assets/167005981/917522fd-6538-487d-86d4-2b07e533ead9)
![IMG_20240415_234427](https://github.com/libretro/Lakka-LibreELEC/assets/167005981/446922e2-d6d8-4123-9036-9b457f309420)

And once everything is done and I can boot on Lakka : 
![Capture d'écran 2024-04-17 130048](https://github.com/libretro/Lakka-LibreELEC/assets/167005981/18bc64c9-041c-4c9d-b3c1-4881dc688201)
Storage size appears to be 26 MB.

And here's what I've on GPARTED : 
![20240417_13h07m37s_grim](https://github.com/libretro/Lakka-LibreELEC/assets/167005981/098755bd-fc07-4a66-9e83-e92126ca255f)
Used : 31.36Mio
Unused : 652.00Kio
Unallocated : 1.82Tio
Size : 1.82Tio

**Comment by ToKe79** [comment]
Unfortunately I don't have a spare 2TB NVMe drive to debug the issue. You have to use manual partition setup.

**Comment by arusabu** [comment]
That's understandable.

That said, strangely enough, checking LAKKA_DISK in gparted solved the problem: it automatically added all the unallocated space to the partition.

Thanks you for your help and the fix.

**Comment by ToKe79** [comment]
@arusabu this image should fix the issue: https://nightly.builds.lakka.tv/members/lakkabuildbot/fix_resize/RPi5.aarch64/

**Comment by arusabu** [comment]
@ToKe79 I confirm it does fix this issue.

There is no file system corruption and resizing is correctly performed on initial boot.
![Capture d'écran 2024-04-18 212706](https://github.com/libretro/Lakka-LibreELEC/assets/167005981/e7b4734e-7baf-4c98-ace0-d923a5f5be07)

Thank you very much for your support.

**Comment by TechnologyClassroom** [comment]
I was going to report this issue and workaround, but it looks like it is fixed. I am seeing this issue on regular 128GB MicroSD cards with the RPi5 Lakka 5.0 img.

This fix alone should probably be reason for a 5.1 release as most people do not know their way around gparted.

Edit: Adding more information.

After the recovery and everything the storage partition was only 26M.

```
/dev/mmcblk0p2           26.0M      2.2M     23.1M   9% /storage
```

These two commands from Ubuntu fixed it for me without gparted:

```
sudo e2fsck -f /dev/sdb2
e2fsck 1.46.5 (30-Dec-2021)
Pass 1: Checking inodes, blocks, and sizes
Pass 2: Checking directory structure
Pass 3: Checking directory connectivity
Pass 3A: Optimizing directories
Pass 4: Checking reference counts
Pass 5: Checking group summary information

LAKKA_DISK: ***** FILE SYSTEM WAS MODIFIED *****
LAKKA_DISK: 197/8192 files (0.5% non-contiguous), 2105/8192 blocks


sudo resize2fs /dev/sdb2
resize2fs 1.46.5 (30-Dec-2021)
Resizing the filesystem on /dev/sdb2 to 30009344 (4k) blocks.
The filesystem on /dev/sdb2 is now 30009344 (4k) blocks long.
```

Then it shows the full 113G:

```
/dev/sdb2               113G  2.2M  113G   1% /mnt
```
